scripts/clang-tidy.sh

```diff
@@ -0,0 +1,96 @@
+    if ! command -v bear >/dev/null 2>&1; then
+        cat <<EOF >&2
+The Bear utility needs to be installed (e.g. sudo apt install bear, brew install
+bear, etc.). The Bear utility creates compile_commands.json from Autotools builds.
+https://github.com/rizsotto/Bear
+EOF
+        exit 2
+    fi
+    if [[ ! -e build/config.status ]]; then
+        echo "The build subdirectory needs to be created, autogen.sh run, and configure run." >&2
+        exit 2
+    fi
+    cd build
+    rm -f clang-tidy.out compile_commands.json
+    make -j clean
+    bear -- make -j
+fi
+
+if [[ ${fix} -ne 0 ]]; then
+    clang_tidy_fix="--fix --fix-errors --fix-notes"
+    njobs=1
+else
+    clang_tidy_fix=""
+    njobs=0
+fi
+
+if [[ -n ${checks} ]]; then
+    # Many clang-analyzer-* checks have to be individually disabled. If wildcard -clang-analyzer-*
+    # is used, clang-tidy will fail with "no checks" even if additional checks appear after them.
+    clang_tidy_checks="-clang-analyzer-apiModeling.*,-clang-analyzer-core.*,-clang-analyzer-cplusplus.*,-clang-analyzer-deadcode.*,-clang-analyzer-fuchsia.*,-clang-analyzer-nullability.*,-clang-analyzer-optin.*,-clang-analyzer-osx.*,-clang-analyzer-security.*,-clang-analyzer-unix.*,-clang-analyzer-valist.*,-clang-analyzer-webkit.*,${checks}"
+else
+    # These are the default checks for --picky. Even for --picky, many of them are disabled.
+    clang_tidy_checks="bugprone-*,performance-*,modernize-*,readability-redundant-*,-modernize-use-trailing-return-type,-modernize-avoid-c-arrays,-modernize-use-override,-modernize-use-nodiscard,-modernize-macro-to-enum,-performance-enum-size,-modernize-use-auto,-bugprone-easily-swappable-parameters,-bugprone-reserved-identifier,google-explicit-constructor,misc-static-assert,misc-header-include-cycle,misc-definitions-in-headers,-modernize-return-braced-init-list,-clang-analyzer-cplusplus.NewDeleteLeaks"
+
+    if [[ ${picky} -eq 0 ]]; then
+        # This disables more checks when --picky is not used.
+        clang_tidy_checks="${clang_tidy_checks},-readability-redundant-inline-specifier,-modernize-use-equals-default,-performance-avoid-endl,-modernize-pass-by-value,-modernize-use-default-member-init,-readability-redundant-string-init,-modernize-use-equals-delete,-performance-unnecessary-copy-initialization,-readability-redundant-access-specifiers,-modernize-use-bool-literals,-readability-redundant-member-init,-performance-unnecessary-value-param,-bugprone-narrowing-conversions"
+    fi
+fi
+
+cd build
+output=$(realpath clang-tidy.out)
+
+echo "Using clang-tidy --checks=${clang_tidy_checks}"
+
+perl -ne 'print "$1\0" if /"file":\s*"([^"]+)"/ && !m:/external/:' compile_commands.json | \
+    xargs -0 -P "${njobs}" -I {} timeout 30 sh -c 'echo "Checking {}"; clang-tidy '"${clang_tidy_fix}"' --quiet --format-style=file -p . --system-headers=0 --checks="'"${clang_tidy_checks}"'" "{}"' | tee "${output}" || true
+
+echo "Results are in ${output}"
```

Comment:
The `clang-tidy.sh` script has `clang-tidy` checks optimized for SST Core, by not needlessly listing too many warnings which are false positives, but allowing `--picky` to increase the number of warnings, while still being informative and not too much noise.